controllers/pkg/giteaclient/giteaclient.go

```diff
@@ -36,10 +37,25 @@ type GiteaClient interface {
 	Get() *gitea.Client
 }
 
-func New(client resource.APIPatchingApplicator) GiteaClient {
-	return &gc{
-		client: client,
+var lock = &sync.Mutex{}
+
+var singleInstance *gc
+
+func New(client resource.APIPatchingApplicator, ctx context.Context) GiteaClient {
+	if singleInstance == nil {
```

Comment:
is it not better to have a flag like started and if set the started is not called

Reply:
Very similar to the original code, `Start(ctx)` is called only once when the instance is created for first time as can be seen in lines from L48 to L51 unless I misunderstood your comment.

Reply:
it is a flavour of how I do it, but your approach is ok